From 56208d4cba23496c09f69961a32b1b9662f06aef Mon Sep 17 00:00:00 2001
From: Tobias Waldekranz <tobias@waldekranz.com>
Date: Mon, 22 Apr 2024 23:18:01 +0200
Subject: [PATCH 25/25] net: dsa: mv88e6xxx: Honor ports being managed via
 in-band-status
Organization: Addiva Elektronik

Keep all link parameters in their unforced states when the port is
declared as being managed via in-band-status, and let the MAC
configure itself, via whatever method is being used (e.g. SGMII or
USXGMII autoneg)

This state is the default set up by mv88e6xxx_port_setup_mac(), so all
we have to do is to make the phylink MAC callbacks no-ops in cases
when in-band-status is being used.

Signed-off-by: Tobias Waldekranz <tobias@waldekranz.com>
---
 drivers/net/dsa/mv88e6xxx/chip.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/dsa/mv88e6xxx/chip.c b/drivers/net/dsa/mv88e6xxx/chip.c
index 76bb8b492ca9..29691280cff4 100644
--- a/drivers/net/dsa/mv88e6xxx/chip.c
+++ b/drivers/net/dsa/mv88e6xxx/chip.c
@@ -885,6 +885,9 @@ static void mv88e6xxx_mac_link_down(struct dsa_switch *ds, int port,
 	const struct mv88e6xxx_ops *ops;
 	int err = 0;
 
+	if (mode == MLO_AN_INBAND)
+		return;
+
 	ops = chip->info->ops;
 
 	mv88e6xxx_reg_lock(chip);
@@ -915,6 +918,9 @@ static void mv88e6xxx_mac_link_up(struct dsa_switch *ds, int port,
 	const struct mv88e6xxx_ops *ops;
 	int err = 0;
 
+	if (mode == MLO_AN_INBAND)
+		return;
+
 	ops = chip->info->ops;
 
 	mv88e6xxx_reg_lock(chip);
-- 
2.34.1

